exes/ethercat/inc/public/tfc/ec/devices/device.hpp

```diff
@@ -45,6 +47,51 @@ using devices_list = devices_type<
 // https://b.atch.se/posts/constexpr-meta-container/
 // https://mc-deltat.github.io/articles/stateful-metaprogramming-cpp20
 
+template <typename manager_client_t>
+using device_variant = std::variant<abt::easyecat<manager_client_t>,
+                                    beckhoff::ek1100,
+                                    beckhoff::el1002<manager_client_t>,
+                                    beckhoff::el1008<manager_client_t>,
+                                    beckhoff::el1809<manager_client_t>,
+                                    beckhoff::el2794<manager_client_t>,
+                                    beckhoff::el2004<manager_client_t>,
+                                    beckhoff::el2008<manager_client_t>,
+                                    beckhoff::el2809<manager_client_t>,
+                                    beckhoff::eq2339<manager_client_t>,
+                                    // beckhoff::el3054,
+                                    beckhoff::el4002,
+                                    schneider::atv320::device<manager_client_t>,
+                                    schneider::lxm32m<manager_client_t>,
+                                    eilersen::e4x60a::e4x60a<manager_client_t>,
+                                    default_device>;
```

Comment:
Really not sure with this still.

one benefit over monostate could be to expose information over dbus for the unimplemented
devices. for example the name, product code and generic information.

Reply:
default device could also in the future support devices via runtime, with some kind of configuration

Reply:
That could be useful, or simply just getting pdo sizes, reading and writing random sdo location. Let's stick with this.